### PR TITLE
Fix migrations and preserved User table and Tasks features

### PR DESCRIPTION
**Pull request by KellyAH**
# Problem

Running migrations on the latest main branc is currently broken and throws an error when running `rails db:migrate`.

```
get-things-done-app/db/migrate/20201024193303_create_thoughts.rb:5: syntax error, unexpected tSYMBEG, expecting end t.has_one :user_category ^ 
get-things-done-app/db/migrate/20201024193303_create_thoughts.rb:13: syntax error, unexpected end, expecting end-of-input
```

# Cause
Lailson says as a Rails best practice, **do NOT edit old migration scripts**. Just make new migration scripts that makes the changes you want. Migrations broke because db/migrate/20201024193303_create_thoughts.rb was edited in commit https://github.com/KellyAH/get-things-done-app/commit/7c9d3fbe6b0897e07eac6f2a45a53bee6df89db3. Which threw the above error. 

And other errors were thrown complaining about priority failing to be created because it already existed when the migration scripts (which I deleted in this commit https://github.com/KellyAH/get-things-done-app/pull/9/commits/86b5f0b22ca75704eec22a6bcdb8472f52dc3d6d) were executed. They failed because they were trying to add a priority column to the thoughts table when it was already added by `20201024193303_create_thoughts.rb`. 

# Reference
[Rails Active Record Migrations guide](http://guides.rubyonrails.org/migrations.html)
https://stackify.com/rails-migration-a-complete-guide/

# Code Changes
This branch was cut off of `practice-rails-stuff` branch (see code changes in https://github.com/KellyAH/get-things-done-app/pull/6) which also had the latest main merged into it. So besides migration fixes and a tiny alert message feature, this branch **also contains**:
- all the features https://github.com/KellyAH/get-things-done-app/pull/6 had 
- commit https://github.com/KellyAH/get-things-done-app/commit/7c9d3fbe6b0897e07eac6f2a45a53bee6df89db3 which was merged to main Nov 27th.

**Structural and Behavioral**

**https://github.com/KellyAH/get-things-done-app/pull/6 Features**
- Make submit from on home page create new Thoughts
- Create `Tasks` data and display it on the homepage
- add the ability to destroy and edit `Tasks`
- Add ability to turn `Thoughts` into a `Tasks`

**Fixed migration scripts**
* Fixed the broken db/migrate/20201024193303_create_thoughts.rb and added ability to create a new thought when a user object doesn't exist. Because the app currentlyy doesn't have the capability to select a user on the Frontend. And creating a new thought was breaking without my change.  (see https://github.com/KellyAH/get-things-done-app/pull/9/commits/75e6d88f1d8cdc823a5e89364b72ecd60cc44971)
* deleted no longer needed migration scripts (see https://github.com/KellyAH/get-things-done-app/pull/9/commits/86b5f0b22ca75704eec22a6bcdb8472f52dc3d6d)

**New Feature**
* Added a red error message that displays the Thought object error when the app fails to create a Thought on the home page.

red error message

# Tests
Rebuilt DB successfully on my local `rails db:drop db:create db:migrate` and all main features to create tasks, thoughts etc work when manually testing the app on my local.

app still works

### CODE REVIEW

**Review by karolina-benitez — APPROVED** [review]

File: app/controllers/tasks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  before_action :set_task, only: [:show, :edit, :update, :destroy]
+
+
+  def new
+    @task = Task.new(priority: "medium")
+    if params[:thought_id]
+      thought = Thought.find(params[:thought_id])
+      @task.description = thought.description
+    end
+  end
```
Comment (KellyAH) [inline app/controllers/tasks_controller.rb:11]: change to 

```suggestion
  def KDFHKHKDFKDF
    @task = Task.new(priority: "medium")
    if params[:thought_id]
      thought = Thought.find(params[:thought_id])
      @task.description = thought.description
    end
  end
```

File: app/views/homepage/index.html.erb [hunk 2]
```diff
@@ -1,14 +1,12 @@
...
-</br>
-
-<%= form_with(url: "/thoughts", method: "post") do %>
-  <%= label_tag(:q, "Write 3 or more things on your mind now.") %>
-  </br>
-  <%= text_field_tag(:q) %>
-  <%= submit_tag("Submit") %>
+<%= form_with model: @new_thought do |form| %>
+  <%= form.text_field :description %>
+  <%= form.button "Create" %>
```
Comment (KellyAH) [inline app/views/homepage/index.html.erb:9]: dsadsdsas

File: app/views/homepage/index.html.erb [hunk 3]
```diff
@@ -1,14 +1,12 @@
+<p id="notice"><%= notice %></p>
+<p id="alert"><%= alert %></p>
+
 <h1>Welcome to The Get Things Done App.</h1>
```
Comment (naomiquinones) [inline app/views/homepage/index.html.erb]:
```suggestion
<h1>Welcome to The Get Things Done App.</h1>

<p id="notice"><%= notice %></p>
<p id="alert"><%= alert %></p>

```

File: app/controllers/thoughts_controller.rb [hunk 4]
```diff
@@ -12,11 +12,6 @@ def index
   def show
   end
 
-  # GET /thoughts/new
-  def new
-    @thought = Thought.new
-  end
-
   # GET /thoughts/1/edit
   def edit
```
Comment (NOI-lgtm) [inline app/controllers/thoughts_controller.rb:17]: This is empty method

**Review by karolina-benitez — COMMENTED** [review]
LGTM!

**Review by karolina-benitez — CHANGES_REQUESTED** [review]
BLAH BLAH

File: app/assets/stylesheets/scaffolds.scss [hunk 5]
```diff
@@ -36,6 +36,9 @@ div.actions {
 #notice {
   color: green; }
 
+#alert {
+  color: red; }
```
Comment (naomiquinones) [inline app/assets/stylesheets/scaffolds.scss]:
```suggestion
#alert {
  color: #cc1234;
  font-weight: bold; }
```

File: app/controllers/thoughts_controller.rb [hunk 6]
```diff
@@ -12,11 +12,6 @@ def index
   def show
```
Comment (NOI-lgtm) [inline app/controllers/thoughts_controller.rb:13]:
```suggestion
  # Delegating to find tasks id into private methods set_thought
  def show
```

**Review by naomiquinones — CHANGES_REQUESTED** [review]
I added css

**Review by NOI-lgtm — COMMENTED** [review]
Add comment in empty method

**Comment by NOI-lgtm** [comment]
LGTM! 👍

**Review by NOI-lgtm — APPROVED** [review]
Amazing 💯 👍

**Review by karolina-benitez — APPROVED** [review]
👸🏻

**Review by naomiquinones — APPROVED** [review]

**Pull request merged**